src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -397,6 +350,23 @@ typedef struct {
     char kvsname[MPIDI_KVSAPPSTRLEN];
     char pname[MPI_MAX_PROCESSOR_NAME];
     int port_name_tag_mask[MPIR_MAX_CONTEXT_MASK];
+
+    /* Capability settings */
+#ifdef MPIDI_OFI_ENABLE_RUNTIME_CHECKS
```

Comment:
Ack. I remembered another problem with this method that prevented me from doing it before. OFI returns more than one item for a single provider. If you go do `fi_info` on a system with libfabric installed, you'll get back a whole bunch of sockets providers and some might have different capabilities.

I think we can get around it by using this as the template. Assume you pass this in on the command line:

```
./configure --with-device=ch4:ofi:sockets,psm2,runtime
```

We'll capture the first one that isn't `runtime` in your list at configure time and store it in a macro somewhere. If that macro matches one of the known providers, we'll preload the CVARs with the values from those providers. We'll have to be careful not to overwrite the user's input here so we'll probably have to change the CVARs to be something that isn't a valid value that we can use as a placeholder.

After all that, if the user wants to make a small tweak to an existing provider's capability set, they can overwrite a single CVAR without having to do all of them.